infra/src/Pos/Infra/Diffusion/Subscription/Common.hs

```diff
@@ -211,8 +210,6 @@ networkSubscribeTo' logTrace oq bucket nodeType peersVar keepalive subStates sen
 type SubscriptionMessageConstraints =
     ( Message MsgSubscribe
     , Message MsgSubscribe1
-    , Bi MsgSubscribe
-    , Bi MsgSubscribe1
```

Comment:
A bit confused about this part, just for the sake of my own understanding (since code builds & tests pass, I assume it's correct). Is the below statement accurate?

> `SubscriptionMessageConstraints` only exists because of having Orphan instances. It asserts that, at the location where it is used, there is are orphan `Bi MsgSubscribe` & `Bi MsgSubscribe1` instances in scope. Since we have moved the `Bi` instances to where the types are defined, `SubscriptionMessageConstraints` no longer needs to include them.

Reply:
Yes, you're correct. That's exactly accurate.

Reply:
Yes, having orphan instances was causing extra constraints to be required in other parts of the code. This is what makes orphan instances a lose/lose situation.